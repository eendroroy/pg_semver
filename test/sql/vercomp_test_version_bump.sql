LOAD 'vercomp';

SELECT VERSION_BUMP('1.1.1', 0);
SELECT VERSION_BUMP('1.1.1', 1);
SELECT VERSION_BUMP('1.1.1', 2);

SELECT VERSION_BUMP('1.1.1-alpha.1', 0);
SELECT VERSION_BUMP('1.1.1-alpha.1', 1);
SELECT VERSION_BUMP('1.1.1-alpha.1', 2);

SELECT VERSION_BUMP('1.1.1-rc1*', 1);
SELECT VERSION_BUMP('1.1.1', -1);
SELECT VERSION_BUMP('1.1.1', 3);
