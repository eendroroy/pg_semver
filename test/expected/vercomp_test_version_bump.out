LOAD 'vercomp';
SELECT VERSION_BUMP('1.1.1', 0);
 version_bump 
--------------
 2.1.1
(1 row)

SELECT VERSION_BUMP('1.1.1', 1);
 version_bump 
--------------
 1.2.1
(1 row)

SELECT VERSION_BUMP('1.1.1', 2);
 version_bump 
--------------
 1.1.2
(1 row)

SELECT VERSION_BUMP('1.1.1-alpha.1', 0);
 version_bump  
---------------
 2.1.1-alpha.1
(1 row)

SELECT VERSION_BUMP('1.1.1-alpha.1', 1);
 version_bump  
---------------
 1.2.1-alpha.1
(1 row)

SELECT VERSION_BUMP('1.1.1-alpha.1', 2);
 version_bump  
---------------
 1.1.2-alpha.1
(1 row)

SELECT VERSION_BUMP('1.1.1-rc1*', 1);
ERROR:  Invalid Version '1.1.1-rc1*'.
LINE 1: SELECT VERSION_BUMP('1.1.1-rc1*', 1);
                            ^
DETAIL:  Version '1.1.1-rc1*' contains invalid character.
HINT:  Did you mean '1.1.1-rc1'?
SELECT VERSION_BUMP('1.1.1', -1);
ERROR:  Invalid bump number '-1'.
DETAIL:  Bump number should be between 0 and 2.
HINT:  Use bump number between 0 and 2.
SELECT VERSION_BUMP('1.1.1', 3);
ERROR:  Invalid bump number '3'.
DETAIL:  Bump number should be between 0 and 2.
HINT:  Use bump number between 0 and 2.
